sql/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -227,8 +227,24 @@ message Expression {
     }
 
     message Struct {
-      DataType struct_type = 1;
-      repeated Literal elements = 2;
+      // (Deprecated) The data type of the struct. Only needs to be set if elements field is used.
+      // Consider using fields instead.
+      optional DataType struct_type = 1 [deprecated = true];
+
+      // (Deprecated) The elements of the struct. Consider using fields instead.
+      repeated Literal elements = 2 [deprecated = true];
+
+      repeated StructField fields = 3;
+
+      message StructField {
+        string name = 1;
```

Comment:
Thank you for this improvement. So we will contain the field name in every element, the space complexity would be O(n) compared to O(1) in the deprecated way? So what's the perf gain from this change?

Reply:
On second thought, it is not necessary to include the field name in every element, since not setting the data_type field should be sufficient for inferring types.